### PR TITLE
docs: improve need for speed intro

### PR DESCRIPTION
**Pull request by jaimem88**
fixes https://github.com/exercism/go/issues/1589

### CODE REVIEW

**Comment by jaimem88** [comment]
@iHiD @ErikSchierboom ready for review 😄

File: exercises/concept/need-for-speed/.docs/introduction.md [hunk 1]
```diff
@@ -1,18 +1,46 @@
 # Introduction
 
-In Go, a `struct` is a sequence of named elements called _fields_, each field having a name and type. The name of a field must be unique within the struct. `Structs` can be compared with the _class_ in the Object Oriented Programming paradigm.
+In Go, a `struct` is a sequence of named elements called _fields_, each field has a name and type. The name of a field must be unique within the struct.
+`Structs` can be compared with _classes_ in the Object-Oriented Programming paradigm.
 
-You create a new struct by using the `struct` keyword, a **_built-in type_**, and explicitly define the name and type of the fields as shown in the example below.
+You create a new struct by using the **_built-in type_** keywords `type` and `struct`, and explicitly define the name and type of the fields.
```
Comment (junedev) [inline exercises/concept/need-for-speed/.docs/introduction.md:6]: This now reads "using the built-in type keywords type ...". I think it should say "Using the `type` keyword and the built-in type `struct` ..."

File: exercises/concept/need-for-speed/.docs/introduction.md [hunk 2]
```diff
@@ -1,18 +1,46 @@
...
 
-Struct fields are accessed using a `.` notation.
+Once you have defined the `struct`, you need to create a new instance defining the fields _in order_, for example:
 
 ```go
-someStruct.someField = "someValue"
+person := Person{
+	"John",
+	20,
+}
```
Comment (junedev) [inline exercises/concept/need-for-speed/.docs/introduction.md:22]: As this syntax is not really recommended as it creates brittle code (breaks when a field is added) I would recommend only showing it in the about.md of the struct concept.

**Review by junedev — CHANGES_REQUESTED** [review]
Thanks for the update! I left two comments. Besides addressing those it would be great if you could also copy your new introduction as introduction of the structs concept.

Furthermore the about.md should contain the info from the introduction and whatever is mentioned there additionally that is not yet in the introduction.

File: exercises/concept/need-for-speed/.docs/introduction.md [hunk 3]
```diff
@@ -1,18 +1,46 @@
...
+
+```go
+fmt.Printf("name: %s age: %d\n", person.name, person.age)
+// Output: name: John age: 22
+```
+
+Instance fields can be modified by accessing each field separately:
+
+```go
+anotherPerson.name = "Mary"
```
Comment (ErikSchierboom) [inline exercises/concept/need-for-speed/.docs/introduction.md:45]: Maybe shorten this slightly to:

````suggestion
To read or modify instance fields, use the `.` notation:

```go
// Update the age
person.age = 23

fmt.Printf("name: %s age: %d\n", person.name, person.age)
// Output: name: John age: 23
```

Instance fields can be modified by accessing each field separately:
````

File: exercises/concept/need-for-speed/.docs/introduction.md [hunk 4]
```diff
@@ -1,18 +1,46 @@
 # Introduction
 
-In Go, a `struct` is a sequence of named elements called _fields_, each field having a name and type. The name of a field must be unique within the struct. `Structs` can be compared with the _class_ in the Object Oriented Programming paradigm.
+In Go, a `struct` is a sequence of named elements called _fields_, each field has a name and type. The name of a field must be unique within the struct.
+`Structs` can be compared with _classes_ in the Object-Oriented Programming paradigm.
```
Comment (ErikSchierboom) [inline exercises/concept/need-for-speed/.docs/introduction.md:4]: Structs is not a literal type name here.

```suggestion
Structs can be compared with _classes_ in the Object-Oriented Programming paradigm.
```

File: exercises/concept/need-for-speed/.docs/introduction.md (hunk 2 above)
Comment (ErikSchierboom) [reply exercises/concept/need-for-speed/.docs/introduction.md:22]: Sounds like a good idea. The exemplar solution also only uses the named field syntax.

**Review by ErikSchierboom — COMMENTED** [review]
Thanks! Some minor comments.

File: exercises/concept/need-for-speed/.docs/introduction.md (hunk 2 above)
Comment (jaimem88) [reply exercises/concept/need-for-speed/.docs/introduction.md:22]: Sounds good, I'll take care of the exemplar solution too :)

**Comment by jaimem88** [comment]
Thanks! I'll also try to get back to this soon!

File: concepts/structs/about.md [hunk 5]
```diff
@@ -1,39 +1,71 @@
...
+```go
+// Update the age
+person.age = 23
+fmt.Printf("name: %s age: %d\n", person.name, person.age)
+// Output: name: John age: 23
+```
 
-// create a new variable
-var dog Animal
+Fields that don't have an initial value assigned, will have their zero value. For example:
```
Comment (ErikSchierboom) [inline concepts/structs/about.md:34]: Perfect content for the `about.md` document.

File: concepts/structs/about.md [hunk 6]
```diff
@@ -1,39 +1,71 @@
...
-// assign a values to each field by using the . operator
-dog.name = "Rex"
-dog.age = 10
+```go
+person:= Person{}
+fmt.Printf("name: %s age: %d\n", person.name, person.age)
+// Output: name: age: 0
+```
+
+You can create an instance of a `struct` without using the field names, as long as you define the fields _in order_:
```
Comment (ErikSchierboom) [inline concepts/structs/about.md:42]: Again, perfect content for the `about.md` document.

**Review by ErikSchierboom — APPROVED** [review]
Excellent!

**Review by junedev — APPROVED** [review]

**Pull request merged**